api/src/main/java/org/cardanofoundation/rosetta/common/util/ParseConstructionUtil.java

```diff
@@ -310,13 +291,14 @@ private static String getStakingCredentialHex(Operation certOperation) {
     return hex;
   }
 
-  public static Operation parsePoolCertToOperation(Integer network, Certificate cert, Long index,
+  public static Operation parsePoolCertToOperation(Long networkProtocolMagic, Certificate cert, Long index,
       String type)
       throws CborSerializationException, CborException {
     Operation operation = Operation.builder()
         .operationIdentifier(new OperationIdentifier(index, null))
         .type(type)
         .status("")
+        .metadata(new OperationMetadata())
```

Comment:
Why it's here?

Reply:
Because of we can get NPE later
```
if (type.equals(OperationType.POOL_RETIREMENT.getValue())) {
      PoolRetirement poolRetirementCert = (PoolRetirement) cert;
      if (!ObjectUtils.isEmpty(poolRetirementCert)) {
        operation.getMetadata().setEpoch((int) poolRetirementCert.getEpoch());
      }
```